### PR TITLE
Spell reformat

### PR DESCRIPTION
**Pull request by hppeng-wynn**
Change internal repr. of spells to use new schema (more info, more general, slightly easier to work with)

Removed all spells except melee for now...

Misc:
- Fixed gangel mult
- Fix raw damage calculation being completely broken

### CODE REVIEW

File: js/damage_calc.js [hunk 1]
```diff
@@ -173,6 +176,125 @@ function calculateSpellDamage(stats, weapon, conversions, use_spell_damage, igno
...
+    type:           "heal"          Internal use
+    name:           str             Display name of part. Should be human readable
+    heal_amount:    num             floating point HP healed (self)
+}
+
+*/
+
+const default_spells = {
+    wand: [{
+        name: "Magic Strike",  // TODO: name for melee attacks?
```
Comment (ferricles) [inline js/damage_calc.js:243]: what

File: js/damage_calc.js [hunk 2]
```diff
@@ -173,6 +176,125 @@ function calculateSpellDamage(stats, weapon, conversions, use_spell_damage, igno
...
+spell_heal: {
+    type:           "heal"          Internal use
+    name:           str             Display name of part. Should be human readable
+    heal_amount:    num             floating point HP healed (self)
+}
+
+*/
+
+const default_spells = {
+    wand: [{
```
Comment (ferricles) [inline js/damage_calc.js:242]: this goes for all the other weapon names - shouldn't these be strings?

File: js/damage_calc.js (hunk 1 above)
Comment (ferricles) [reply js/damage_calc.js:243]: are these just for flavortext when we display them? I'd probably just go "Wand Melee" or "Mage Melee" so we avoid confusion with actual Wynn names

File: js/display.js [hunk 3]
```diff
@@ -1610,44 +1611,48 @@ function displaySpellDamage(parent_elem, overallparent_elem, stats, spell, spell
...
-            let totalDamNormal = _results[0];
-            let totalDamCrit = _results[1];
-            let results = _results[2];
-            
-            for (let i = 0; i < 6; ++i) {
-                for (let j in results[i]) {
-                    results[i][j] = results[i][j].toFixed(2);
-                }
-            }
+        if (damage_info.type === "damage") {
```
Comment (ferricles) [inline js/display.js:1652]: why would it be called damage_info if it wasn't doing damage

File: js/damage_calc.js (hunk 1 above)
Comment (hppeng-wynn) [reply js/damage_calc.js:243]: fair

File: js/damage_calc.js (hunk 2 above)
Comment (hppeng-wynn) [reply js/damage_calc.js:242]: js is not json -- it works

File: js/display.js (hunk 3 above)
Comment (hppeng-wynn) [reply js/display.js:1652]: because me dumb

**Pull request merged**